### ISSUE TITLE
Documentation request - synchronous function in async

### ISSUE THREAD

**Issue by askhogan** [caolan/async#629]
I recently ran into an issue with async.eachSeries where I ran into a maximum call stack exceeded.  There are several closed issues related to this within the issues sections.  Most of them come with github members writing the following comment

"Dont call a synchronous function in async"  

or the program maintainer @caolan writing

"I've added some code to call async.nextTick if it detects a synchronous iterator which should avoid a stack overflow in these cases. But seriously, STOP CALLING SYNC FUNCTIONS IN ASYNC! ...make your functions consistently synchronous or consistently asynchronous instead ;)"

I understand that over time people consistently doing the same behavior is annoying, which leads to the all caps type response.  However, for people running into this issue, my guess is that they didn't start out the day saying to themselves, "I really love calling sync functions in async", my guess is they were probably drawn to node by how fast it was and the ability to operate in async with good intentions.  They probably wrote some methods, placed them in async, got the error, and then when they read the common response they had no idea what you were talking about, because this response doesn't give any examples (and there is nothing in the pitfalls section about this very important rule)   It would be helpful if some examples of this behavior were cited in the Common Pitfalls section. 

In the error I am receiving, obviously I am somehow calling a sync function in async (even after stepping through all code and not realizing how).   Each one of the methods that I use, uses async callbacks.  I do have a nested eachSeries in one of the methods (so this may be it).   All of the guides on the net say to use process.nextTick() quite often, but I have read comments where @colan says this is unnecessary with async... so I am a bit stumped whether I should be converting all my callbacks to process.nextTick() to make them async or what I should be doing.  

tl;dr - please add code examples of using async.eachSeries or async.each and what calling a sync function in async behavior is.

**Comment by aearly** [comment]
BTW, if you have a function that sometimes calls it's callback on the same tick, you can wrap your function with [dezalgo](https://github.com/npm/dezalgo) or [acomb.ensureAsync](https://github.com/aearly/acomb#ensureAsync) to work around this.

**Comment by aearly** [comment]
`ensureAsync` will be released as part of v1.1.0, since we are not using it internally in a few places!
